src/Voron/Data/ITree.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Voron.Data
+{
+    public interface ITree
+    {
+        Slice Name { get; }
```

Comment:
For that matter, we can rely on the usual ToString() method, and avoid the complexity here entire, no?